SFDigitalServices/sf-dahlia-web
# DAH-2972 fix: use h2 for no lottery card heading

## Description
Adds an h2 as a child for the Card component. 

## Jira ticket

https://sfgovdt.jira.com/browse/DAH-2972
## Checklist before requesting review

### Version Control


### Code quality


### Review instructions


### Request review


## Review Instructions
- Go to a FCFS Sales Listing
- Use WAVE to ensure that there are no heading level warnings for the "No Lottery Required" heading
